src/commands/deploy/index.ts

```diff
@@ -1,72 +1,162 @@
-import { Command, Flags } from "@oclif/core";
-import { execSync } from "node:child_process";
+import { Command, Flags, CliUx } from "@oclif/core";
 import { readFileSync, writeFileSync } from "node:fs";
 import path = require("node:path");
+import { CodePromise, Abi } from "@polkadot/api-contract";
+import { Keyring } from "@polkadot/keyring";
+import { readJSONSync } from "fs-extra";
+import { cryptoWaitReady } from "@polkadot/util-crypto";
+import { ChainApi } from "../../lib/substrate-api";
+import { KeyringPair } from "@polkadot/keyring/types";
+import { Listr } from "listr2";
+import { ensureSwankyProject } from "../../lib/command-utils";
 export class DeployContract extends Command {
   static description = "Deploy contract to a running node";
 
   static flags = {
+    account: Flags.string({
+      required: true,
+      options: ["Alice", "Bob", "Charlie", "Dave", "Eve", "Ferdie"],
+    }),
+    contract: Flags.string({ char: "c", required: true }),
     gas: Flags.string({
       required: true,
       char: "g",
     }),
     args: Flags.string({
       required: true,
       char: "a",
+      multiple: true,
     }),
   };
 
   static args = [];
 
   async run(): Promise<void> {
+    ensureSwankyProject();
     const { flags } = await this.parse(DeployContract);
-    let config: { contracts: string[] | Record<string, string> } = {
-      contracts: [""],
-    };
-    try {
-      const file = readFileSync("swanky.config.json", { encoding: "utf-8" });
-      config = JSON.parse(file);
-    } catch {
-      throw new Error("No 'swanky.config.json' detected in current folder!");
-    }
 
-    const output = execSync(
-      `cargo contract instantiate --constructor new --args ${flags.args} --suri //Alice --gas ${flags.gas}`,
+    const tasks = new Listr([
+      {
+        title: "Initialising",
+        task: async () => {
+          await cryptoWaitReady();
+        },
+      },
+      {
+        title: "Getting WASM",
+        task: (ctx) => {
+          const buildPath = path.resolve(
+            "contracts",
+            flags.contract,
+            "target",
+            "ink"
+          );
+          const abi = readJSONSync(
+            path.resolve(buildPath, "metadata.json")
+          ) as Abi;
+          const wasm = readFileSync(
+            path.resolve(buildPath, `${flags.contract}.wasm`)
+          );
+          ctx.abi = abi;
+          ctx.wasm = wasm;
+        },
+      },
       {
-        stdio: "pipe",
-        cwd: path.resolve(
-          "contracts",
-          typeof config.contracts[0] === "string"
-            ? config.contracts[0]
-            : config.contracts[0].name
-        ),
-      }
-    ).toString("utf-8");
+        title: "Getting keypair",
+        task: (ctx) => {
+          const keyring = new Keyring({ type: "sr25519" });
+          const pair = keyring.createFromUri(`//${flags.account}`);
```

Comment:
It seems like this creates a new account by using the provided account name as the seed instead of loading from a json account backup file?

Reply:
same as above.